src/SSLClientImpl.cpp

```diff
@@ -476,6 +478,7 @@ int SSLClientImpl::m_run_until(const unsigned target) {
 /* see SSLClientImpl.h*/
 unsigned SSLClientImpl::m_update_engine() {
     const char* func_name = __func__;
+    uint8_t retry = 5;
```

Comment:
What problem does this variable solve? It seems like it re-runs the state machine, however I believe SSLClient should do that anyway.

Reply:
When using STM32Ethernet the for loop never exits because STM32Ethernet write always returns an error. This happens on close so it might be due to some race condition on writing to a closed/closing socket. 

```
            wlen = get_arduino_client().write(buf, len);
```

Reply:
I made a modification to replace the retry variable with better error handling, does this still fix the issue?